### ISSUE TITLE
NMS ONNX convert to tensorrt engine error

### ISSUE THREAD

**Issue by jinfagang** [pytorch/vision#1620]
I have found that, torchvision::nms does not support batchsize, but onnx definition had definite a batchsize dim,

that is to say, we have to introduce several unsqueeze op when export.

this is impossible to convert to tensorrt engine, since we will got error like this:

```
----------------------------------------------------------------
Input filename:   model_nms.onnx
ONNX IR version:  0.0.4
Opset version:    11
Producer name:    pytorch
Producer version: 1.3
Domain:           
Model version:    0
Doc string:       
----------------------------------------------------------------
WARNING: ONNX model has a newer ir_version (0.0.4) than this parser was built against (0.0.3).
Parsing model
While parsing node number 0 [Unsqueeze -> "4"]:
ERROR: /onnx-tensorrt/builtin_op_importers.cpp:2035 In function importUnsqueeze:
[8] Assertion failed: axis != BATCH_DIM

```

Which means, onnx-tensorrt doesn't support any operation on the first dimension (batch dimension).

Does there any way to make it work on tensorrt?

```
import torch
import numpy as np
from torch import nn
import onnxruntime
from torchvision.ops import RoIAlign
from torchvision.ops import nms


# register_custom_op()


torch.manual_seed(1024)
np.random.seed(1024)

model_p = 'model_nms.onnx'


class TinyModel(nn.Module):
    def __init__(self):
        super(TinyModel, self).__init__()
        self.expander = nn.Conv1d(100, 1, 1, 1)
        self.roi_boxes = roi_boxes
        self.atom = torch.tensor([[0.22], [0.14], [0.1], [0.4]]).to(device)

    def forward(self, x: torch.Tensor):
        rois = x['rois']
        score = x['scores']
        x = nms(rois, score, 0.6)
        return x


def export_onnx():
    model = TinyModel().to(device)
    rois = torch.rand(100, 4).to(device)
    scores = torch.rand(100).to(device)
    sample_input = {
        'rois': rois,
        'scores': scores
    }
    model.eval()
    torch.onnx.export(model, sample_input, model_p, input_names=[
                      'rois', 'scores'], output_names=['values'], opset_version=11)
    print('onnx model exported. forward now...')
    # forward now

    out_pt = model(sample_input)
    print('pytorch output: {}'.format('-'*80))
    print(out_pt)

    rois = rois.cpu().numpy()
    scores = scores.cpu().numpy()
    ort_session = onnxruntime.InferenceSession(model_p)
    ort_inputs = {ort_session.get_inputs()[0].name: rois,
                  ort_session.get_inputs()[1].name: scores, }
    out_onnx = ort_session.run(None, ort_inputs)
    print('onnx output: {}'.format('-'*80))
    print(out_onnx)


if __name__ == "__main__":
    export_onnx()

```

code attached to reproduce problem.

**Comment by fmassa** [comment]
@lara-hdr do you know where the issue is?

**Comment by fmassa** [comment]
@jinfagang did you solve the issue? What was it?

**Comment by jinfagang** [comment]
@fmassa  It was supported in ONNX 11.

However I thought torchvision does nms slightly different from onnx definition?
onnx definition used score_threshold, does torchvision have a default score_threshold?

The model trace out to onnx inference on onnxruntime have different output than original pytorch

**Comment by fmassa** [comment]
Torchvision version of nms has a score threshold. There are unit tests comparing the output of ONNX and torchvision, so they should match. If that's not the case, please give us a minimum reproducible example illustrating the problem

**Comment by jinfagang** [comment]
Yeah, sure. actually when I implement a tensorrt nms plugin which parses nms node traced from torchvision I found even definition slight different than onnx.

onnx have inputs scores with num_classes dim, and returns cls_indices, box_indices, batch_indices, while seems torchvision directly using a box: [98, 4], score: [98] as input (say 98 is num boxes).

**Comment by fmassa** [comment]
The implementation should still functionally perform the same thing. So the indices should still match.

**Comment by fmassa** [comment]
FYI, here is the NMS test that checks that the outputs are the same: https://github.com/pytorch/vision/blob/cec7ea727ab5905ee6d7039a4f1b980124684879/test/test_onnx.py#L77-L86

**Comment by ledinhtri97** [comment]
Hello, jinfagang

I run your script export nms_onnx. And output pt model and onnx model are different? Did you fixed and can convert NMS to tensorRT?
